core/lib/libtomcrypt/src/tee_ltc_provider.c

```diff
@@ -465,7 +465,11 @@ static TEE_Result hash_final(void *ctx, uint32_t algo, uint8_t *digest,
 #if defined(_CFG_CRYPTO_WITH_ACIPHER)
 
 #define LTC_MAX_BITS_PER_VARIABLE   (4096)
+#ifdef CFG_CRYPTO_ECC
+#define LTC_VARIABLE_NUMBER         (80)
+#else
 #define LTC_VARIABLE_NUMBER         (50)
+#endif
```

Comment:
Yes, ECC requires lots of variables. 50 is not enough.

I think we should think of a way to optimize it because we request, for ECC, 80 temporary variables (which is ok) of size 4096 bits (which is way too more). This could be part of another patch.